File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第8章等差数列.py
#2019年第十届省赛真题
#等差数列  lanqiaoOJ 192
from  math import *
n=int(input())
a=list(map(int,input().split()))
a.sort()
d=0
for i in range(1,n):  d=gcd(d,a[i]-a[i-1])
if d==0:  print(n)
else:     print((a[-1]-a[0])//d+1)

File: 配套资源/配套代码/程序设计竞赛专题挑战教程-代码/第8章等差数列.cpp
//2019年第十届省赛真题
//等差数列  lanqiaoOJ 192

#include<bits/stdc++.h>
using namespace std;
int a[100000];
int main(){
    int n;   cin>>n;
    for(int i=0;i<n;i++)   cin>>a[i];
    sort(a,a+n);
    int d=0;
    for(int i=1;i<n;i++)   d = __gcd(d,a[i]-a[i-1]);
    if(d==0) cout<<n<<endl;
    else     printf("%d\n",(a[n - 1] - a[0]) / d + 1);
    return 0;
}
